src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationService.java

```diff
@@ -49,25 +57,73 @@ public class NocNotificationService {
+            handleClaimantNocEmails(caseDetailsNew, partyName);
+        } else {
+            // send respondent noc change email
+            handleRespondentNocEmails(caseDetailsPrevious, caseDetailsNew, changeRequest);
+            partyName = NocNotificationHelper.getRespondentNameForNewSolicitor(changeRequest, caseDataNew);
         }
 
-        String newOrgId = changeRequest.getOrganisationToAdd().getOrganisationID();
-        sendEmailToNewOrgAdmin(newOrgId, caseDetailsNew, partyName);
+        // send organisation noc change email
+        handleOrganisationNocEmails(caseDataPrevious, caseDetailsNew, changeRequest, partyName, currentUserEmail);
 
+        // send tribunal noc change email
         sendTribunalEmail(caseDataPrevious);
+    }
+
+    public void handleClaimantNocEmails(CaseDetails caseDetailsNew, String partyName) {
+        CaseData caseDataNew = caseDetailsNew.getCaseData();
+
+        // send respondent or respondent solicitor noc change email
+        caseDataNew.getRespondentCollection().forEach(resp -> {
+            Map<String, String> emailAddressesMap =
+                    getRespondentAndRespRepEmailAddressesMap(caseDataNew, resp);
+
+            emailAddressesMap.forEach((email, respondentId) -> {
+                String caseLink = StringUtils.isNotBlank(respondentId)
+                        ? emailService.getSyrCaseLink(caseDetailsNew.getCaseId(), respondentId)
+                        : emailService.getExuiCaseLink(caseDetailsNew.getCaseId());
+                emailService.sendEmail(claimantTemplateId, email,
+                        buildPersonalisationWithPartyName(caseDetailsNew, partyName, caseLink));
+            });
+        });
+
+        // send claimant noc change email
+        String claimantEmail = NotificationHelper.getEmailAddressForClaimant(caseDataNew);
+
+        if (isNullOrEmpty(claimantEmail)) {
+            log.warn("missing claimantEmail");
+            return;
+        }
+
+        Map<String, String> personalisation = buildRespondentPersonalisation(caseDetailsNew, partyName);
+        emailService.sendEmail(respondentTemplateId, claimantEmail, personalisation);
+    }
+
+    public void handleRespondentNocEmails(CaseDetails caseDetailsPrevious,
+                                          CaseDetails caseDetailsNew,
+                                          ChangeOrganisationRequest changeRequest) {
+
+        CaseData caseDataNew = caseDetailsNew.getCaseData();
+        CaseData caseDataPrevious = caseDetailsPrevious.getCaseData();
+        String partyName = NocNotificationHelper.getRespondentNameForNewSolicitor(changeRequest, caseDataNew);
+        // send claimant or claimant solicitor noc change email
+        if (!isClaimantNonSystemUser(caseDataPrevious) || isRepresentedClaimantWithMyHmctsCase(caseDataPrevious)) {
+            sendClaimantEmail(caseDetailsPrevious, caseDetailsNew, partyName);
+        }
```

Comment:
I think this logic needs to be amended.
If a citizen submits a case and then say an offline rep is added for the citzen, this logic will still be true as the claimant is still a system user albeit they have representation. sendClaimantEmail then can send an email to the rep even though they are offline as we aren't checking to see if the rep is a system user in that method